.clang-format

```diff
@@ -0,0 +1,24 @@
+
```

Comment:
Hi @haslersn ,

I've separately pushed a slightly different  `.clang-format` file to master (with some reformatting that I wanted to do), 
which makes the format part of this PR no longer needed.

However, the Nix part is still useful.  Please could you abandon this PR and raise a single PR just for the Nix additions?  You can re-use https://github.com/xsco/libdjinterop/pull/1 if you like.

Thanks!

Reply:
Then just merge #1 and close #2